src/components/AccordianNavMenu.tsx

```diff
@@ -0,0 +1,73 @@
+import { Icon } from "@canonical/react-components";
+import { FC, LinkHTMLAttributes, ReactNode, useState } from "react";
+import { useLocation } from "react-router-dom";
+import NavLink from "./NavLink";
+
+interface Props {
+  to: string;
+  baseUrl: string;
+  title: string;
+  children: ReactNode;
+  navItems: ({
+    to: string;
+    title: string;
+    label: string;
+    activeUrlMatches?: string[];
+  } & LinkHTMLAttributes<HTMLElement>)[];
+}
+
+const AccordianNavMenu: FC<Props> = ({
+  baseUrl,
+  title,
+  children,
+  navItems,
```

Comment:
I'd recommend changing the interface, so the secondaryNavLinks are passed as children. Then we don't need to come up with an abstraction of a list of NavLink components, but we can pass them directly from the parent.

I think we need one extra prop for the icon name, but we drop the "to" prop, so in total we don't increase the interface size.